plane/src/proxy/proxy_service.rs

```diff
@@ -117,9 +117,10 @@ impl RequestHandler {
             Ok(response) => Ok(response),
             Err(err) => {
                 let (status_code, body) = match err {
-                    ProxyError::InvalidConnectionToken => {
-                        (hyper::StatusCode::FORBIDDEN, "Forbidden")
-                    }
+                    ProxyError::InvalidConnectionToken => (
+                        hyper::StatusCode::GONE,
+                        "The backend is no longer available or the connection token is invalid.",
```

Comment:
Is there any way to differentiate between these two cases?

Reply:
Unfortunately not reliably, because we only keep connection tokens around until they expire and we would need to keep them around indefinitely. We could encode the expiration time in the token, I suppose, and then for (supposedly) unexpired tokens we would know that it was invalid if we did not find it.